api/v1/ratelimitpolicy_types.go

```diff
@@ -153,6 +153,9 @@ func (p *RateLimitPolicy) Kind() string {
 // +kubebuilder:validation:XValidation:rule="!(has(self.defaults) && has(self.limits))",message="Implicit and explicit defaults are mutually exclusive"
 // +kubebuilder:validation:XValidation:rule="!(has(self.defaults) && has(self.overrides))",message="Overrides and explicit defaults are mutually exclusive"
 // +kubebuilder:validation:XValidation:rule="!(has(self.overrides) && has(self.limits))",message="Overrides and implicit defaults are mutually exclusive"
+// +kubebuilder:validation:XValidation:rule="!(has(self.overrides) || has(self.defaults)) ? has(self.limits) && size(self.limits) > 0 : true",message="At least one spec.limits most be defined"
+// +kubebuilder:validation:XValidation:rule="has(self.overrides) ? has(self.overrides.limits) && size(self.overrides.limits) > 0 : true",message="At least one spec.overrides.limits most be defined"
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:rule="!has(self.overrides) || has(self.overrides.limits) && size(self.overrides.limits) > 0",message="At least one spec.overrides.limits most be defined"
```

Just a matter of personal taste, up to you.

Reply:
The ternary operator makes more sense to me. The not has always trips me up. I tend not to like to have navigate checks.

Reply:
LGTM